homeassistant/components/logger/__init__.py

```diff
@@ -103,6 +111,16 @@ def _set_log_level(logger, level):
     getattr(logger, "orig_setLevel", logger.setLevel)(LOGSEVERITY[level])
 
 
+def _add_log_filter(logger, patterns):
+    """Add a Filter to the logger based on a regexp of the filter_str."""
+
+    def pattern_to_filter(p):
```

Comment:
Logging is performance critical.  Please

- Precompile all your regexs with `re.compile`
- Match them all in one filter instead of multiple

Reply:
1. I had the same though, but using cv.is_regex in on L45 (to my surprise) has pre-compiled them!
2. Great point.

I do a lot of performance-related work at my 9 to 5 so I did take perf into consideration. This approach has no impact except for on the services that it is enabled. The impact on those services is bounded by their log volume and the complexity of the patterns.